cloud-arm/src/main/java/com/sequenceiq/cloudbreak/cloud/arm/ArmTemplateBuilder.java

```diff
@@ -0,0 +1,97 @@
+    @Autowired
+    private ArmClient armClient;
+
+    public String build(String stackName, CloudCredential cloudCredential, CloudStack cloudStack) {
+        try {
+            String imageName = cloudStack.getImage().getImageName();
+            imageName = imageName.replace("https://", "");
+            String[] split = imageName.split("/");
+            ArmCredentialView armCredentialView = new ArmCredentialView(cloudCredential);
+            Map<String, Object> model = new HashMap<>();
+            model.put("storage_account_name", armClient.getStorageName(cloudCredential, cloudStack.getRegion()));
+            model.put("image_storage_container_name", ArmSetup.IMAGES);
+            model.put("storage_container_name", ArmSetup.VHDS);
+            model.put("storage_vhd_name", split[2]);
+            model.put("admin_user_name", "cloudbreak");
+            model.put("stackname", stackName);
+            model.put("ssh_key", armCredentialView.getPublicKey());
+            model.put("region", CloudRegion.valueOf(cloudStack.getRegion()).value());
+            model.put("subnet1Prefix", cloudStack.getNetwork().getSubnet().getCidr());
+            model.put("addressPrefix", cloudStack.getNetwork().getParameter("subnetCIDR", String.class));
+            model.put("groups", cloudStack.getGroups());
+            model.put("ports", cloudStack.getSecurity().getRules().get(0).getPorts());
+            model.put("port_protocol", cloudStack.getSecurity().getRules().get(0).getProtocol());
+            model.put("corecustomData",
+                    new String(
+                            Base64.encodeBase64(
+                                    String.format("%s", cloudStack.getImage().getUserData(InstanceGroupType.CORE))
+                                            .getBytes())));
+            model.put("gatewaycustomData",
+                    new String(
+                            Base64.encodeBase64(
+                                    String.format("%s", cloudStack.getImage().getUserData(InstanceGroupType.GATEWAY))
+                                            .getBytes())));
+            String generatedTemplate = processTemplateIntoString(freemarkerConfiguration.getTemplate(armTemplatePath, "UTF-8"), model);
+            LOGGER.debug("Generated Heat template: {}", generatedTemplate);
+            return generatedTemplate;
+        } catch (IOException | TemplateException e) {
+            throw new RuntimeException("Failed to process the Arm TemplateBuilder", e);
+        }
+    }
+
+    public String buildParameters(CloudCredential credential, Network network, Image image) {
+        try {
+            return processTemplateIntoString(freemarkerConfiguration.getTemplate(armTemplateParametersPath, "UTF-8"),  new HashMap<>());
+        } catch (IOException | TemplateException e) {
+            throw new RuntimeException("Failed to process the Arm TemplateParameterBuilder", e);
+        }
+    }
+
+    @VisibleForTesting
```

Comment:
@Visible for testing shall be avoided, and can be used with a Context config + Mocking and inside the SomeApplicationContext shall contain a Freemarker config 

+@RunWith(SpringJUnit4ClassRunner.class)
+@SpringApplicationConfiguration(classes = SomeApplicationContext.class)
+public class LaunchStackComponentTest {